src/example-types/metricGroups/pivot.js

```diff
@@ -0,0 +1,116 @@
+let F = require('futil')
+let _ = require('lodash/fp')
+let { getStats } = require('./stats')
+let { getField } = require('../../utils/fields')
+let types = require('../../../src/example-types')
+let { transmuteTree } = require('../../utils/futil')
+let { simplifyBucket } = require('../../utils/elasticDSL')
+
+let lookupTypeProp = (prop, type) => _.get(`${type}GroupStats.${prop}`, types)
+
+// PivotTable -> Query:
+//  rows -> columns -> values
+//  (same) values for ALL group levels
+// Rows/Columns are buckets, values are metrics
+
+let aggsForValues = (node, schema) =>
+  _.flow(
+    // Add `pivotMetric-` to auto keys so we can skip camelCasing it in the response
+    _.keyBy(
+      ({ key, type, field }) =>
+        key || F.compactJoin('-', ['pivotMetric', type, field])
+    ),
+    _.mapValues(({ key, type, field, ...props }) => ({
```

Comment:
This is a very interesting lint error - while `key` is not used in the function body, it _is_ important. Omitting `key` here would include it in the `props` spread and then pass it along as part of the body of the ES aggregation. Might need an eslint-disable comment here 🤔